File: src/main/java/app/askresume/global/jwt/dto/JwtTokenDto.java
package app.askresume.global.jwt.dto;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.util.Date;

public record JwtTokenDto(
        String grantType,

        String accessToken,

        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
        Date accessTokenExpireTime,

        String refreshToken,

        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
        Date refreshTokenExpireTime
) {

    public JwtTokenDto(String grantType, String accessToken, Date accessTokenExpireTime, String refreshToken, Date refreshTokenExpireTime) {
        this.grantType = grantType;
        this.accessToken = accessToken;
        this.accessTokenExpireTime = accessTokenExpireTime;
        this.refreshToken = refreshToken;
        this.refreshTokenExpireTime = refreshTokenExpireTime;
    }
}

File: src/main/kotlin/app/askresume/global/jwt/dto/JwtTokenDto.kt
package app.askresume.global.jwt.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.util.*

data class JwtTokenDto(
    val grantType: String,

    val accessToken: String,

    @field:JsonFormat(
        shape = JsonFormat.Shape.STRING,
        pattern = "yyyy-MM-dd HH:mm:ss",
        timezone = "Asia/Seoul"
    )
    val accessTokenExpireTime: Date,

    val refreshToken: String,

    @field:JsonFormat(
        shape = JsonFormat.Shape.STRING,
        pattern = "yyyy-MM-dd HH:mm:ss",
        timezone = "Asia/Seoul"
    )
    val refreshTokenExpireTime: Date,
)

